File: apps/android/app/src/main/java/com/barns/app/domain/repository/CatalogRepository.kt
package com.barns.app.domain.repository

import com.barns.app.domain.model.CatalogItem

/**
 * Official, read-only catalog content. Mock now; a real read-only API may
 * back this later without changing use cases or screens.
 */
interface CatalogRepository {
    suspend fun catalogItems(): List<CatalogItem>
    suspend fun catalogItem(id: String): CatalogItem?
}

File: apps/ios/Barns/BarnsApp/Domain/Repositories/CatalogRepository.swift
/// Official, read-only catalog content. Mock now; a real read-only API may
/// back this later without changing use cases or screens.
protocol CatalogRepository {
    func catalogItems() async throws -> [CatalogItem]
    func catalogItem(id: CatalogItem.ID) async throws -> CatalogItem?
}
